src/main/java/com/microsoft/sqlserver/jdbc/ThreePartName.java

```diff
@@ -0,0 +1,71 @@
+package com.microsoft.sqlserver.jdbc;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+class ThreePartName {
+    /*
+     * Three part names parsing For metdata calls we parse the procedure name into parts so we can use it in sp_sproc_columns sp_sproc_columns
+     * [[@procedure_name =] 'name'] [,[@procedure_owner =] 'owner'] [,[@procedure_qualifier =] 'qualifier']
+     *
+     */
+    private static final Pattern THREE_PART_NAME = Pattern.compile(JDBCSyntaxTranslator.getSQLIdentifierWithGroups());
+
+    private final String databasePart;
+    private final String ownerPart;
+    private final String procedurePart;
+
+    private ThreePartName(String databasePart, String ownerPart, String procedurePart) {
+        this.databasePart = databasePart;
+        this.ownerPart = ownerPart;
+        this.procedurePart = procedurePart;
+    }
+
+    public String getDatabasePart() {
```

Comment:
Thanks. What I meant was let's remove the public for now as this is used inside the driver source.